src/library_tty.js

```diff
@@ -97,7 +97,7 @@ mergeInto(LibraryManager.library, {
           if (ENVIRONMENT_IS_NODE) {
             // we will read data by chunks of BUFSIZE
             var BUFSIZE = 256;
-            var buf = new Buffer(BUFSIZE);
+            var buf = Buffer.alloc ? Buffer.alloc(BUFSIZE) : new Buffer(BUFSIZE);
```

Comment:
is this for a specific version of node? which?

Reply:
@kripken 
I'm on v10.6.0.  I see the warning when I run the code from #6505.

Reply:
I see, thanks.

Ok, this looks like a good fix then. However, I see we have similar code in `src/library_nodefs.js`, there is a `bufferFrom` method there. I think we should move that helper function to a more general place, like `support.js` (the optimizer will remove it if it isn't used), and use it from both of those locations. Maybe also the name should be changed, to something like `nodeJSBufferFrom`?